hpat/distributed.py

```diff
@@ -73,10 +79,38 @@
 
 
 class DistributedPass(object):
-    """analyze program and transfrom to distributed"""
+    """The summary of the class should be here for example below is the summary line for this class
+    
+    This class analyzes program and transforms to distributed
+    """
 
     def __init__(self, func_ir, typingctx, targetctx, typemap, calltypes,
                  metadata):
+        """ Here are moer details about the init method or the class 
+       
+        The details can be multiple lines. You can mention all the supported
+        and not supported args here or you can do it in ``Parameters`` section
+        
+        Note
```

Comment:
Is it possible to extract `Note` and other sections into one documentation page and only `Note` to an other documentation page? I mean a differences between `user` and `developer` documentation we discussed off line.

Reply:
I couldn't find something like that in Sphinx. @fschlimb do you know if Sphinx has a way to do this? If Sphinx can't do that then I'll have to write a parsing script for that.

Reply:
What about
https://sublime-and-sphinx-guide.readthedocs.io/en/latest/conditions.html
or
https://www.sphinx-doc.org/en/master/usage/extensions/ifconfig.html
?

Reply:
This is interesting! I will try it out. Thanks @fschlimb!